src/ExpressiveInstaller/Resources/src/ExpressiveSymfonyContainer.php

```diff
@@ -0,0 +1,226 @@
+<?php
+
+namespace App;
+
+use Symfony\Component\DependencyInjection\Container;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Dumper\PhpDumper;
+use Symfony\Component\DependencyInjection\Reference;
+
+/**
+ * Configuration for the Symfony Dependency Injection container.
+ *
+ * This class provides functionality for the following service types:
+ *
+ * - Aliases
+ * - Delegators
+ * - Factories
+ * - Invokable classes
+ * - Services (known instances)
+ */
+class ExpressiveSymfonyContainer
+{
+    const ENABLE_CACHE = 'cache_symfony_container';
+
+    /**
+     * @var array
+     */
+    private $config;
+
+    /**
+     * @var string|null
+     */
+    private $cachedContainerFile;
+
+    /**
+     * @param array $config
+     */
+    public function __construct(
+        array $config,
+        $cachedContainerFile = null
+    ) {
+        $this->config = $config;
+        $this->cachedContainerFile = $cachedContainerFile;
+    }
+
+    /**
+     * return Container
+     */
+    public function create()
+    {
+        $loadFromCache = null !== $this->cachedContainerFile &&
+            file_exists($this->cachedContainerFile) &&
+            isset($this->config[static::ENABLE_CACHE]) &&
+            $this->config[static::ENABLE_CACHE];
+
+        if ($loadFromCache) {
+            return $this->loadFromCache($this->cachedContainerFile);
+        }
+
+        $container = $this->build();
+
+        $this->cacheContainer($container);
+
+        return $container;
+    }
+
+    /**
+     * @var string $cachedContainerFile
+     * @return \ProjectServiceContainer
+     */
+    private function loadFromCache($cachedContainerFile)
+    {
+        require_once $cachedContainerFile;
+
+        $container = new \ProjectServiceContainer();
+
+        // Register the container in the container
+        // In this way in can be referenced in factories
+        $container->set('container', $container);
+
+        // Inject config
+        $container->set('config', $this->config);
+
+        // Inject services
+        if (! empty($this->config['dependencies']['services'])
+            && is_array($this->config['dependencies']['services'])
+        ) {
+            foreach ($this->config['dependencies']['services'] as $name => $service) {
+                $container->set($name, $service);
+            }
+        }
+
+        return $container;
+    }
+
+    /**
+     * @return ContainerBuilder
+     */
+    private function build()
+    {
+        // Build container
+        $container = new ContainerBuilder();
+
+        // Register the container in the container
+        // In this way in can be referenced in factories
+        $container->register('container')->setSynthetic(true);
```

Comment:
this does not make sense. Symfony already exposes the container as a service (named `service_container`), and optimizes the dumping. If you want to have it available as `container`too, create an alias instead